src/bridge/crypto/rsa.cc

```diff
@@ -107,28 +134,38 @@ int Sign(int type, const unsigned char *digest_bytes,
   // conversion functions refer to some OpenSSL API functions.
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const CryptoKeyHandle *crypto_key_handle,
-      GetCryptoKeyHandleFromOpenSslRsa(rsa), false);
+      GetCryptoKeyHandleFromOpenSslRsa(rsa), 0);
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const DigestCase digest_type,
-      ConvertOpenSslNidToDigestType(type), false);
+      ConvertOpenSslNidToDigestType(type), 0);
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const std::string digest_string,
-      MakeDigestString(digest_bytes, digest_length), false);
+      MakeDigestString(digest_bytes, digest_length), 0);
 
   // Delegate handling of the signing operation to the backing layer.
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const std::string signature,
-      crypto_key_handle->Sign(digest_type, digest_string), false);
+      crypto_key_handle->Sign(digest_type, digest_string), 0);
 
   // Copy results into the return pointers.
+  if (signature_length != nullptr) {
+    *signature_length = signature.length();
+  }
   if (signature_return != nullptr) {
+    // Sanity check on `RSA_size` and the length of the signature.
+    if (static_cast<int>(signature.length()) > RSA_size(rsa)) {
+      KMSENGINE_SIGNAL_ERROR(
+          Status(StatusCode::kFailedPrecondition,
```

Comment:
This error message seems a bit verbose. Simply stating the the signature is too long would be sufficient, I think.